README.md

```diff
@@ -353,3 +353,59 @@ or support is available at `quay.io/compliance-operator/must-gather:latest`:
 ```
 $ oc adm must-gather --image=quay.io/compliance-operator/must-gather:latest
 ```
+
+Metrics
+=======
+The compliance-operator exposes the following metrics to Prometheus when cluster-monitoring is available.
+
+    # HELP compliance_operator_compliance_remediation_status_total A counter for the total number of updates to the status of a ComplianceRemediation
+    # TYPE compliance_operator_compliance_remediation_status_total counter
+    compliance_operator_compliance_remediation_status_total{name="remediation-name",state="NotApplied"} 1
+
+    # HELP compliance_operator_compliance_scan_status_total A counter for the total number of updates to the status of a ComplianceScan
+    # TYPE compliance_operator_compliance_scan_status_total counter
+    compliance_operator_compliance_scan_status_total{name="scan-name",phase="AGGREGATING",result="NOT-AVAILABLE"} 1
+
+    # HELP compliance_operator_compliance_scan_error_total A counter for the total number of encounters of error
+    # TYPE compliance_operator_compliance_scan_error_total counter
+    compliance_operator_compliance_scan_error_total{name="scan-name",error="some_error"} 1
+
+After logging into the console, navigating to Monitoring -> Metrics, the compliance_operator* metrics can be queried using the metrics dashboard. The `{__name__=~"compliance.*"}` query can be used to view the full set of metrics.
+
+Testing for the metrics from the cli can also be done directly with a pod that curls the metrics service. This is useful for troubleshooting.
+
+```
+oc run --rm -i --restart=Never --image=registry.fedoraproject.org/fedora-minimal:latest -n openshift-compliance metrics-test -- bash -c 'curl -ks -H "Authorization: Bea
+rer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" https://metrics.openshift-compliance.svc:8585/metrics-co' | grep compliance
+```
+
+Note: Due to a limitation with the CSV format, when deploying through OLM the `prometheus-k8s` RoleBinding deployed in the operator namespace references a non-existent Role. This results in the compliance-operator metrics not being scraped by the cluster-monitoring-operator.
+
+The cluster-monitoring logs will display an error for the operator namespace:
+```
+$ oc -n openshift-monitoring logs -c prometheus prometheus-k8s-0 | grep openshift-compliance
+...
+level=error ts=2021-09-07T19:09:08.598Z caller=klog.go:96 component=k8s_client_runtime func=ErrorDepth msg="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:429: Failed to watch *v1.Endpoints: failed to list *v1.Endpoints: endpoints is forbidden: User \"system:serviceaccount:openshift-monitoring:prometheus-k8s\" cannot list resource \"endpoints\" in API group \"\" in the namespace \"openshift-compliance\": RBAC: role.rbac.authorization.k8s.io \"prometheus-k8s\" not found"
+```
+
+To fix this, as the cluster-admin, create the prometheus-k8s Role inside of the operator namespace:
+```
```

Comment:
So we can't put the Role under `deploy/` because it would have been picked up into the bundle, right? I'm just wondering if we could put the Role into a manifest under SCM control and instead point to `oc create` invocation which might be a bit more user friendly and maybe even possible to use in a test later on?